packages/neon/neon/lib/src/utils/global_options.dart

```diff
@@ -23,54 +21,43 @@ class GlobalOptions {
     this._sharedPreferences,
     this._packageInfo,
   ) {
-    themeMode.stream.listen((final value) {
-      _themeOLEDAsDarkEnabledSubject.add(value != ThemeMode.light);
-    });
-
-    _pushNotificationsDistributorsSubject.listen((final distributors) async {
-      final allowed = distributors.isNotEmpty;
-      _pushNotificationsEnabledEnabledSubject.add(allowed);
-      if (!allowed) {
-        await pushNotificationsEnabled.set(false);
-      }
-    });
-
-    pushNotificationsEnabled.stream.listen((final enabled) async {
-      if (enabled) {
-        final response = await Permission.notification.request();
-        if (response.isPermanentlyDenied) {
-          _pushNotificationsEnabledEnabledSubject.add(false);
-        }
-        if (!response.isGranted) {
-          await pushNotificationsEnabled.set(false);
-        }
-      } else {
-        await pushNotificationsDistributor.set(null);
+    themeMode.addListener(_themeModeListner);
+    pushNotificationsEnabled.addListener(_pushNotificationsEnabledListener);
+    rememberLastUsedAccount.addListener(_rememberLastUsedAccountListener);
+  }
+
+  void _rememberLastUsedAccountListener() {
+    initialAccount.enabled = !rememberLastUsedAccount.value;
+    if (rememberLastUsedAccount.value) {
+      initialAccount.value = null;
+    } else {
+      // Only override the initial account if there already has been a value,
+      // which means it's not the initial emit from rememberLastUsedAccount
+      initialAccount.value = initialAccount.values.keys.first;
+    }
+  }
+
+  Future<void> _pushNotificationsEnabledListener() async {
+    if (pushNotificationsEnabled.value) {
+      final response = await Permission.notification.request();
+      if (response.isPermanentlyDenied) {
+        pushNotificationsEnabled.enabled = false;
       }
-    });
-
-    rememberLastUsedAccount.stream.listen((final remember) async {
-      _initialAccountEnabledSubject.add(!remember);
-      if (remember) {
-        await initialAccount.set(null);
-      } else {
-        // Only override the initial account if there already has been a value,
-        // which means it's not the initial emit from rememberLastUsedAccount
-        if (initialAccount.hasValue) {
-          await initialAccount.set((await initialAccount.values.first).keys.first);
-        }
+      if (!response.isGranted) {
+        pushNotificationsEnabled.value = false;
       }
-    });
+    } else {
+      pushNotificationsDistributor.value = null;
+    }
+  }
+
+  void _themeModeListner() {
+    themeOLEDAsDark.enabled = themeMode.value != ThemeMode.light;
   }
```

Comment:
I think removing it is better. The behavior is quite inconsistent because it was still enabled for light theme regardless of the resulting theme

Reply:
I mean that's sorta intended.
I might want to switch back to the dark theme and still keep the oled pref.

We could use a tristate CheckBox in this case and display the undetermined state.
Or only render it checked if enabled (both something we need to change in `CheckBoxSettingsTile`).
Maybe with an option?

What do you prefer?


---

For reference the tristate looks like (just imagine it greyed out if it's also disabled):
<img width="1024" alt="unnamed" src="https://github.com/provokateurin/nextcloud-neon/assets/25266387/5c4b787c-d73b-4a91-b1ab-376b3802474d">

Reply:
Ah sorry there was a typo in the comment. I think we can just remove the logic to disable the checkbox if the theme is the light theme. Just make it possible to always change this setting regardless of the selected theme

Reply:
but having a setting that doesn't change anything is a bit bad isn't it?

Reply:
Why wouldn't it change anything? I only want to have the enablement logic removed, not the option or anything else